### PR TITLE
[eglib] Replace strerror with strerror_r as the former is not thread safe.

### PR DESCRIPTION
**Pull request by kumpera**
This change would be nicer if the code was moved to the runtime so we could use our PAL.

### CODE REVIEW

File: eglib/src/gstr.c [hunk 1]
```diff
@@ -191,10 +195,56 @@ g_strdup_printf (const gchar *format, ...)
...
+	if (!error_messages [errnum]) {
+#ifndef G_OS_WIN32
+		pthread_mutex_lock (&strerror_lock);
+#endif
+
+#ifdef HAVE_STRERROR_R
+		char tmp_buff [128]; //Quite arbritrary
+		tmp_buff [0] = 0;
+
+		strerror_r (errnum, tmp_buff, sizeof (tmp_buff) - 1); //Spec is not clean on whether size argument includes space for null terminator or not
```
Comment (luhenry) [inline eglib/src/gstr.c]: you should check for ERANGE, and increase the size of tmp_buff in this case.

Comment (kumpera) [inline eglib/src/gstr.c]: done

**Comment by kumpera** [comment]
Retrying the build to rule out the ARMv5 crash

**Comment by kumpera** [comment]
build

**Comment by luhenry** [comment]
@kumpera what I usually do to rule out single build failure is go to the build on Jenkins and do a "Retry" from their, it avoids rebuilding everything.

**Pull request merged**